### PR TITLE
feat: bottom-bar controls state machine [WM-24]

### PR DESCRIPTION
**Pull request by dougmartin**
## Summary

Rewrites the enable/disable rules for the bottom-bar control buttons (Setup, Spark, Reload, Restart, Start/Stop, Fire Line, Helitack) to follow the simulation's lifecycle (default, setup changed, spark placed, running, paused, ended, restarted, reloaded). Each button now lights up only when its action makes sense, so the parent Hazbot help-overlay epic (AP-80) can rely on a clean visual state machine.

Implementation lands in seven phases plus a follow-up CSS hotfix:

1. `setupChanged` observable on `SimulationModel`, preserved by `restart()`, reset by `reload()`.
2. `simulationEnded` computed (`simulationStarted && !simulationRunning && engine?.fireDidStop`) with a value-tested truth table and an observer-level reactivity test.
3. Per-button computeds on `SimulationModel`: `setupEnabled`, `startEnabled`, `reloadEnabled` (`setupChanged || sparks.length > 0`), `restartEnabled`.
4. Setup wizard snapshot-diff: open-time `useRef` capture, diff-before-mutate in `applyAndClose`, helper extracted to `setup-snapshot.ts` for unit-testable wind-direction coverage.
5. `bottom-bar.tsx` rewired with component-class getters that compose simulation predicates with the `ui.interaction` guard. `handleReload` and `handleRestart` reset `ui.interaction = null`. Full `bottom-bar.test.tsx` rewrite covers the six-state matrix plus edge cases (Paused vs. Ended, authoring gate, ui.interaction reset, drag-to-move spark survives Restart). One `log-events.test.tsx` test patched, one removed (vacuous under new rules).
6. Disabled visual styling per Zeplin: content `opacity: 0.35` on inner span, button background unchanged, `filter: grayscale(1)`. Includes `opacity: 1 !important` override on MUI's `:disabled` so the spec's content opacity isn't compounded with MUI's default `0.25` whole-button fade (caught during browser verification).
7. Browser-level Cypress regression guard at `cypress/e2e/bottom-bar-state-machine.cy.ts` covering all seven states.

Spec captured at [`specs/WM-24-model-controls-states.md`](specs/WM-24-model-controls-states.md).

## Test plan

- [ ] `npm test` passes (387 tests, all green locally)
- [ ] `npm run lint` clean (0 errors)
- [ ] `npm run build` succeeds
- [ ] `npx cypress run --spec "cypress/e2e/bottom-bar-state-machine.cy.ts"` passes all 7 states
- [ ] Walk the seven lifecycle states in the browser; confirm Zeplin matrix
- [ ] Sighted-user check on disabled-button opacity (also try Chrome DevTools "Emulate vision deficiencies" panel); flag the designer (Michael Tirenin) if rendered icons are insufficient
- [ ] Confirm `smoke.cy.ts` and `terrairn-setup.cy.ts` still pass under the new disabled rules

## Notes for reviewers

- The state machine is encoded as `@computed` properties on `SimulationModel` for pure-simulation predicates, plus component-class getters on `BottomBar` for predicates that need `ui.interaction`. The `simulationEnded` computed's reactivity is carried by `simulationRunning`; `engine?.fireDidStop` is a discriminator read only (documented inline at the field).
- "Stop" semantically remains a Pause; the rename is deferred to a future sprint per the designer. No follow-up Jira ticket has been filed yet.
- The closed spec file enumerates 8 explicitly-deferred follow-up items, none with Jira tickets.

### CODE REVIEW

File: src/models/simulation.ts [hunk 1]
```diff
@@ -126,6 +127,31 @@ export class SimulationModel {
     }
   }
 
+  // True when simulationStarted && !simulationRunning && engine.fireDidStop.
+  // Reactivity contract: simulationRunning carries the edge — engine?.fireDidStop
+  // is a discriminator read only. The supported tick() path flips both
+  // (simulation.ts:315-317), so the computed re-evaluates when expected. Future
+  // refactorers: do not rely on fireDidStop driving reactivity directly.
```
[the remark this answers is not in the document]
Comment (dougmartin) [reply src/models/simulation.ts]: Fixed in 776d168 — dropped the brittle `simulation.ts:315-317` reference and reworded the comment to describe the `tick()` invariant (`simulationRunning = false` when `engine.fireDidStop` becomes true) so it won't go stale on future re-numberings.

File: cypress/e2e/bottom-bar-state-machine.cy.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+// machine. Covers each of the seven states by driving the real bottom-bar in
+// a running app, asserting the HTML `disabled` attribute per the Zeplin
+// matrix. Catches full-page reactivity wiring breaks, @observer-decoration
+// regressions, and build-tooling failures that the React-Testing-Library
+// tests in bottom-bar.test.tsx can't.
+//
+// Does NOT cover visual styling regressions (opacity, grayscale) — that's
+// the job of the Step 6 unit tests, which assert the .Mui-disabled rules
+// directly. A future Zeplin-driven visual-regression pass would close that
+// gap; deliberately out of scope here.
```
[the remark this answers is not in the document]
Comment (dougmartin) [reply cypress/e2e/bottom-bar-state-machine.cy.ts]: Fixed in 776d168 — you're right, there are no Jest tests asserting `.Mui-disabled` rules. Reworded the header comment to point at the actual coverage: SCSS rules in `src/components/icon-button.scss` (`&:disabled, &.Mui-disabled`) verified by manual inspection against the Zeplin spec, with an automated visual-regression pass deferred.

**Review by scytacki — APPROVED** [review]
Looks good.
